Add Reptile Dragon 2 build log (#2133)
PX4/PX4-user_guide: en/frames_plane/reptile_dragon_2.md

This emphasis on payload makes this airplane an ideal candidate for a PX4 installation.

## Overview

The goal of this build was to create an efficient, long endurance FPV platform to be used for general PX4 testing and development.

Key airframe features:

- Spacious interior
- Easy access to the entire fuselage cavity with large top hatch
- Rear hatch
- Removable V tail or conventional tail options included
- Threaded inserts in the wings and fuselage top for external mounting
- Numerous mounting features
  - Top antenna hole
  - Top GPS cover
  - Side "T" antenna mounts
  - Rear electronics tray
  - Front facing "action cam" cutout
  - Front facing FPV camera cutout 
- Removable wings
- Low stall speed
- Gentle handling 

Key build features

- Easy overall build
- Easy access to Pixhawk and all peripherals
- FPV with camera pan mount
- Air data from pitot/static probe
- ~40 minute long flight times

## Parts list

- [Reptile Dragon 2 kit](https://usa.banggood.com/REPTILE-DRAGON-2-1200mm-Wingspan-Twin-Motor-Double-Tail-EPP-FPV-RC-Airplane-KIT-or-PNP-p-1805237.html?cur_warehouse=CN&ID=531466)
- [Pixhawk 5x Carrier board](https://shop.holybro.com/pixhawk-5x_p1279.html)

> - [ARK6X FMU](https://arkelectron.com/product/arkv6x/)

Reviewer: @chris1seto Just to be clear, you are using the carrier board for the Pixhawk 5x from Holybro with the AR6X FMU - what's the benefit of that? I mean wouldn't it be easier just to go "all holybro" here? Or was there a cost issue or something else.

It's fine, just first question that came into my head.

Author: Basically what happened was the airplane was originally set up for a 5x carrier and Fmu, but Alex had just finished and sent me one of his first 6x's to test. He hadn't finished the carrier board yet, so the majority of the airplane's life up until now had that combination. He just sent me the finished carrier board, so I plan to edit the build log to reflect that being installed now. So basically, the build log reflects his progress on developing the 6x and carrier board and finishing them at different times,

- [Holybro power module](https://shop.holybro.com/pm02d-power-module_p1285.html)
- [Holybro M9N GPS module](https://shop.holybro.com/holybro-m9n-gps_p1280.html)
- MS4525DO differential pressure module and pitot tube

> - [Caddx Vista FPV air unit](https://caddxfpv.com/products/caddx-vista-kit)

Reviewer: Note, we still need to remember to add the FPV stuff after this and cross link.

Author: Yup, FPV stuff is next on my list after this build log is done

The ELRS RX was attached to the sidewall of the airframe next to the airspeed pressure sensor, using double-sided tape.

### USB

A right angle USB C extension cable was used to allow easy access to the USB C port on the FMU.

![Rear USB cable hatch](../../assets/airframes/fw/reptile_dragon_2/usb_hatch.jpg)

The cable was installed such that it escapes the pixhawk heading towards the aft of the airplane. The cable continues to run to the rear hatch, where the excess length can be securely wound into a knot.
Access to this cable can be accomplished by simply removing the rear hatch and unknotting the cable.

## Software Setup

For this build, we need to use the PX4 modules [crsf_rc](../modules/modules_driver.md#crsf-rc) and [msp_osd](../modules/modules_driver.md#msp-osd).
These require some custom configuration to enable.

First, follow [this guide to setup a development environment](../dev_setup/dev_env.md ) and [this guide to get the PX4 source code](../dev_setup/building_px4.md).

Once a build environment has been setup, open a terminal and `cd` into the `PX4-Autopilot` directory.

> To launch the [PX4 board config tool](../hardware/porting_guide_config.md#px4-menuconfig-setup) run:

Reviewer: Is this the correct link for the tool? i.e. is menuconfig the tool you mean? I always used to edit files if I needed to enable things and haven't played with it since these changes went in.

Reviewer: ```suggestion
To launch the [PX4 board config tool (`menuconfig`)](../hardware/porting_guide_config.md#px4-menuconfig-setup) run:

```

```
make ark_fmu-v6x_default boardconfig`
```

### `crsf_rc` Module

PX4 includes a standalone CRSF parser module which supports telemetry and CRSF LinkStatistics.
To use this module, the default `rc_input` module must be disabled and the `crsf_rc` module must be enabled.

1. In the PX4 board config tool, navigate to the `drivers` submenu, then scroll down to highlight `rc_input`.
2. Use the enter key to remove the `*` from `rc_input` checkbox.
3. Scroll to highlight the `RC` submenu, then press enter to open it.
4. Scroll to highlight `crsf_rc` and press enter to enable it.
5. Save and exit the PX4 board config tool.

### `msp_osd` Module

The `msp_osd` module steams MSP telemetry to a selected serial port.
The Caddx Vista Air Unit supports listening to MSP telemetry and will show the received telemetry values in its OSD (on screen display).

1. In the PX4 board config tool, navigate to the `drivers` submenu, then scroll down to highlight `OSD`.
2. Use the enter key to open the `OSD` submenu

> 3. Scroll down to highlight `msp_osd` and press enter to enable it

Reviewer: Don't we have to rebuild PX4 after doing the config? Then install it - as a custom build? Shouldn't we explain that?

Author: Yes, I will make sure this reflects that the custom build needs to be uploaded to the board too.

The channel mappings for this build are included in the params file linked at the bottom of this page.
I chose to map the channels in order throttle, roll, pitch, yaw, (blank), and flight mode

:::note
ExpressLRS requires AUX1 as an "arming channel".
This arming channel is separate from PX4's arming mechanism and is used to tell the ELRS TX that is can switch into high transmit power.

In the PX4 channel mappings, I simply skip over this channel.
On my transmitter, this channel is set to always be "high", so ELRS is always armed.
:::

### Motor Setup & Prop Installation

The RD2 kit comes with clockwise and counter clockwise propellers for counter rotating motors.
With counter rotating props, the airplane can be set up such that it has no [critical motors](https://en.wikipedia.org/wiki/Critical_engine).

With no critical motors, controllability will be maximized if a motor fails.
The motor direction should be set such that props should turn towards the fuselage on top of the plane.
In other words, if you look at the left motor with the airplane facing away from you, it should spin clockwise while the right motor should spin counter clockwise.

> With the propellers removed, power the airplane up and use the [Actuator](../config/actuators.md) test in QGC to spin up the motors.

Reviewer: Do you think it worth saying either here or in paramters section that once you have installed the params you can preview the actuator geometry configuration?

Somewhat useful for people to see what config you had to do to get this to work.

> If the left or right motor does not spin in the correct direction, swap two of its ESC leads and check it again.

Reviewer: Can't you change the direction in the actuators page?

Author: The ESCs have a throttle range of 1000 ... 2000, so they cannot be reversed in software

### Final Checks

Prior to the first flight, a comprehensive preflight must be conducted.

I recommend checking the following items:

- Sensor calibration (QGC)
  - Mag calibration
  - Accelerometer calibration
  - Airspeed calibration
  - Level horizon calibration
- Check control surface deflection
 - Right stick -> Right aileron goes up, left aileron goes down
 - Left stick -> Left aileron goes up, right aileron goes down
 - Stick back -> elevator goes up
 -Stick forward -> elevator goes down
 - Left rudder -> Rudder goes left
 - Right rudder -> Rudder goes right
- Check Px4 inputs (in `stabilized mode`)
 - Roll right -> Right Aileron goes down
 - Roll left -> Left aileron goes down
 - Pitch up -> Elevator goes down
 - Pitch down -> Elevator goes up

### First Flight

I recommend performing the first takeoff in manual mode.
Because this airplane has no landing gear, you will either need to throw the airplane yourself, or ideally have a helper throw it.
When throwing any airplane, throw at a slightly nose up attitude with full throttle.

It's critical to be ready to give aft stick input to prevent the airplane from impacting the ground if it happens to be trimmed nosedown.

> Once the airplane is successfully airborne, cruise up to an altitude of a few hundred feet and switch to [Acro mode](../flight_modes/acro_fw.md).

Reviewer: Note, it is good to link the modes to the mode docs.

@chris1seto Is this sensible? Acro mode is harder to fly than position mode - why this order?

Author: I wouldn't necessarily say that acro is "harder" to fly than position mode. I've been voicing that Px4 should not discourage people from flying in manual and acro, because I think they are realistically base level skills for a user who is building their onw airplane, and without those two modes, you basically can't safely maiden a new airplane or land/takeoff. Automatic takeoff and landing needs an autonomous flight mode which needs tuning to get to. Chicken and egg problem.

If the airplane is well behaved in Acro mode, switch to [Position mode](https://docs.px4.io/main/en/flight_modes/position_fw.html).

> Optionally, you can use [Autotuning](../config/autotune.md) to tune the airframe.

Reviewer: Note, I suggest autotuning and link to that. People can use the switch if they want.

## Build Results & Performance

Overall, this build was a success.
The RD2 flies well in this configuration and has plenty of room onboard for sensors and additional hardware.

### Performance

- Stall speed: 15mph indicated
- Cruise speed: 35-50mph
- Endurance: ~40 minutes at 28mph

### Videos & Flight Logs

[Demo Flight log](https://review.px4.io/plot_app?log=6a1a279c-1df8-4736-9f55-70ec16656d1e)

FPV video of flight log:

@[youtube](https://www.youtube.com/watch?v=VqNWwIPWJb0&ab_channel=ChrisSeto)

> ### Params

Reviewer: Shouldn't this section go up before first flight? After all, it is untuned at this point?

Author: It's tuned in the video, this is a demo flight after all setup is complete.

Reviewer: @chris1seto No, I mean "don't you need to load these params to complete the configuration"? - without them the vehicle won't fly.

So my question is "shouldn't these parameters appear before here in the document" because you won't get to first flight without them?

  - Mag calibration
  - Accelerometer calibration
  - Airspeed calibration
  - Level horizon calibration
- Check control surface deflection
 - Right stick -> Right aileron goes up, left aileron goes down
 - Left stick -> Left aileron goes up, right aileron goes down
 - Stick back -> elevator goes up
 -Stick forward -> elevator goes down
 - Left rudder -> Rudder goes left
 - Right rudder -> Rudder goes right
- Check Px4 inputs (in `stabilized mode`)
 - Roll right -> Right Aileron goes down
 - Roll left -> Left aileron goes down
 - Pitch up -> Elevator goes down
 - Pitch down -> Elevator goes up

### First Flight

I recommend performing the first takeoff in manual mode.
Because this airplane has no landing gear, you will either need to throw the airplane yourself, or ideally have a helper throw it.
When throwing any airplane, throw at a slightly nose up attitude with full throttle.

It's critical to be ready to give aft stick input to prevent the airplane from impacting the ground if it happens to be trimmed nosedown.
Once the airplane is successfully airborne, cruise up to an altitude of a few hundred feet and switch to [Acro mode](../flight_modes/acro_fw.md).

> If the airplane is well behaved in Acro mode, switch to [Position mode](https://docs.px4.io/main/en/flight_modes/position_fw.html).
> 
> Optionally, you can use [Autotuning](../config/autotune.md) to tune the airframe.

Reviewer: Note, I'm switching these lines around and suggest this is a good time to autotune, not just "you can". Note also that I use relative links.

```suggestion
This is a good time to use [Autotuning](../config/autotune.md) to tune the airframe.

If the airplane is well behaved in _Acro mode_, switch to [Position mode](../flight_modes/position_fw.md).
```
